odbc/src/api/functions.rs

```diff
@@ -868,22 +868,33 @@ fn sql_driver_connect(conn: &Connection, odbc_uri_string: &str) -> Result<MongoC
         }
     }
 
-    let conn_attrs = conn.attributes.read().unwrap();
-    let database = if conn_attrs.current_catalog.is_some() {
-        conn_attrs.current_catalog.as_deref().map(|s| s.to_string())
-    } else {
-        odbc_uri.remove(&["database"])
-    };
-    let connection_timeout = conn_attrs.connection_timeout;
-    let login_timeout = conn_attrs.login_timeout;
+    // when a mongo_connection is being established, we take the connection attributes that could have been sent
+    // prior to the mongo_connection being established and move them into the mongo_connection
+    let database: Option<String>;
+    {
+        let catalog_database = conn
+            .attributes
+            .read()
+            .unwrap()
+            .as_ref()
+            .unwrap()
+            .current_catalog
+            .as_deref()
+            .map(|s| s.to_string());
+        database = if catalog_database.is_some() {
+            catalog_database
+        } else {
+            odbc_uri.remove(&["database"])
+        };
+    }
+    let mut attrs = conn.attributes.write().unwrap().take().unwrap();
+    attrs.current_catalog = database;
```

Comment:
This feels a little verbose, but makes the Rust compiler happy. This thankfully is also the only place that I take a read lock immediately followed by a write lock on the same RwLock, so the only place this extra scope is introduced. I'm thinking about taking one write lock here for efficiency. Thoughts?